File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = dfs(numbers,target,0)
    return answer

def dfs(numbers, target, ans):

    if not numbers:
        return ans + (target == 0)
    
    ans = ans\
        +dfs(numbers[1:], target+numbers[0],ans)\
        +dfs(numbers[1:], target-numbers[0],ans)
    
    return ans

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let ans = 0
    const dfs = (idx, res) => {
        if (idx === numbers.length) {
            if (res === target) ans+=1
            return
        }
        // -= num
        dfs(idx+1, res-numbers[idx])
        // += num
        dfs(idx+1, res+numbers[idx])
    }
    dfs(0,0)
    return ans
}